src/Components/History/index.jsx

```diff
@@ -0,0 +1,31 @@
+// History.jsx
+import React from 'react';
+import './History.scss';
+
+function History({ history, dispatch }) {
+  const handleHistoryClick = (requestParams) => {
+    dispatch({ type: 'SET_DATA', payload: null }); // Clear previous data
+    dispatch({ type: 'SET_HEADERS', payload: new Headers() }); // Clear previous headers
+    dispatch({ type: 'SET_REQUEST_PARAMS', payload: requestParams });
+  };
```

Comment:
You're not really fetching the results of a previous API call here, you're making a new call with the same requestParams.